### PR TITLE
Dark theme

### PR DESCRIPTION
**Pull request by Kabak-Siarhei**
Complete:
- Dark and Light theme switch.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,113 +1,74 @@
...
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
-      ),
-      home: MyHomePage(title: 'Flutter Demo Home Page'),
+    return BlocProvider(
+      create: (context) => ThemeBloc(),
+      child: BlocBuilder<ThemeBloc, ThemeState>(builder: (context, state) {
+        bool value = false;
```
Comment (shcherbuk96) [inline lib/main.dart:15]: Please rename

File: lib/main.dart [hunk 2]
```diff
@@ -1,113 +1,74 @@
...
-  // that it has a State object (defined below) that contains fields that affect
-  // how it looks.
-
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
-
-  final String title;
+class HomePage extends StatefulWidget {
```
Comment (shcherbuk96) [inline lib/main.dart:29]: Could you please move this widget to separate class?

File: lib/main.dart [hunk 3]
```diff
@@ -1,113 +1,74 @@
...
+  _HomePageState createState() => _HomePageState();
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+class _HomePageState extends State<HomePage> {
+  bool value = false;
 
-  void _incrementCounter() {
+  void onSwitchValue(bool value) {
```
Comment (shcherbuk96) [inline lib/main.dart:39]: Method should be below than build method.

1) fields
2) constructor
3) getters
4)build method
5)public methods
6)private methods

File: lib/main.dart [hunk 4]
```diff
@@ -1,113 +1,74 @@
...
-            Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
-            ),
-          ],
-        ),
+        title: Text('WordToGif'),
```
Comment (shcherbuk96) [inline lib/main.dart:50]: Add localization to your app.
It will be your second task

File: lib/main.dart [hunk 5]
```diff
@@ -1,113 +1,74 @@
...
+          PopupMenuButton(
+            itemBuilder: (context) => [
+              PopupMenuItem(
+                value: 1,
+                child: Row(
+                  children: [
+                    Text('Dark theme'),
+                    Switch(
+                      value: value,
+                      onChanged: (value) => onSwitchValue(value),
```
Comment (shcherbuk96) [inline lib/main.dart:61]: Could you refactor this code?
onChanged: onSwitchValue

File: lib/theme/theme_event.dart [hunk 6]
```diff
@@ -0,0 +1,7 @@
+part of 'theme_bloc.dart';
+
+class ThemeEvent {
```
Comment (shcherbuk96) [inline lib/theme/theme_event.dart:3]: It is a bad realization.
You should add the main abstract event class and then you can extend from them

File: lib/theme/theme_state.dart [hunk 7]
```diff
@@ -0,0 +1,8 @@
+part of 'theme_bloc.dart';
+
+class ThemeState {
```
Comment (shcherbuk96) [inline lib/theme/theme_state.dart:3]: The same issue

File: pubspec.yaml [hunk 8]
```diff
@@ -28,6 +28,7 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
+  flutter_bloc: ^7.3.1
```
Comment (shcherbuk96) [inline pubspec.yaml:31]: I suggest to add the freezed package and then change states and models to freeze classes

File: android/app/src/main/res/values-night/styles.xml [hunk 9]
```diff
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <!-- Theme applied to the Android Window while the process is starting when the OS's Dark Mode setting is on -->
+    <!-- theme applied to the Android Window while the process is starting when the OS's Dark Mode setting is on -->
```
Comment (shcherbuk96) [inline android/app/src/main/res/values-night/styles.xml:3]: It is necessary?

File: android/app/src/main/res/values-night/styles.xml [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
     <style name="LaunchTheme" parent="@android:style/Theme.Black.NoTitleBar">
         <!-- Show a splash screen on the activity. Automatically removed when
              Flutter draws its first frame -->
         <item name="android:windowBackground">@drawable/launch_background</item>
     </style>
-    <!-- Theme applied to the Android Window as soon as the process has started.
+    <!-- theme applied to the Android Window as soon as the process has started.
```
Comment (shcherbuk96) [inline android/app/src/main/res/values-night/styles.xml:9]: And this

File: android/app/src/main/res/values/styles.xml [hunk 11]
```diff
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <!-- Theme applied to the Android Window while the process is starting when the OS's Dark Mode setting is off -->
+    <!-- theme applied to the Android Window while the process is starting when the OS's Dark Mode setting is off -->
```
Comment (shcherbuk96) [inline android/app/src/main/res/values/styles.xml:3]: And this

File: lib/theme/theme_bloc.dart [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import 'package:bloc/bloc.dart';
+
+part 'theme_event.dart';
+part 'theme_state.dart';
+
+class ThemeBloc extends Bloc<ThemeEvent, ThemeState> {
+  ThemeBloc() : super(ThemeState(false)) {
+    on<ThemeEvent>((event, emit) {
```
Comment (shcherbuk96) [inline lib/theme/theme_bloc.dart:8]: Could we change flutter bloc to older version, where we should use yield?

File: lib/theme/theme_bloc.dart [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import 'package:bloc/bloc.dart';
```
Comment (shcherbuk96) [inline lib/theme/theme_bloc.dart:1]: I suggest to add separate bloc folder

File: lib/main.dart (hunk 5 above)
Comment (bagasayn) [reply lib/main.dart:61]: I removed this method. There is also a problem with animation.
https://github.com/flutter/flutter/issues/47865

File: android/app/src/main/res/values-night/styles.xml (hunk 9 above)
Comment (bagasayn) [reply android/app/src/main/res/values-night/styles.xml:3]: No. Another IDE generated this code. I will watch changes in the future.

**Pull request merged**